pkg/scalers/azure_blob_scaler.go

```diff
@@ -81,12 +82,13 @@ func parseAzureBlobMetadata(config *ScalerConfig) (*azureBlobMetadata, kedav1alp
 		return nil, "", fmt.Errorf("no blobContainerName given")
 	}
 
-	if val, ok := config.TriggerMetadata["blobDelimiter"]; ok && val != "" {
-		meta.blobDelimiter = val
-	}
+	meta.blobDelimiter = config.TriggerMetadata["blobDelimiter"]
 
 	if val, ok := config.TriggerMetadata["blobPrefix"]; ok && val != "" {
-		meta.blobPrefix = val + meta.blobDelimiter
+		meta.blobPrefix = val
+		if !strings.HasSuffix(meta.blobPrefix, defaultBlobDelimiter) {
+			meta.blobPrefix += defaultBlobPrefix
+		}
```

Comment:
as mentioned on the issue #1789 

> One complication: it seems that the scaler was appending `blobDelimiter` to `blobPrefix`, which isn't right. To avoid having a breaking change, I append the default `blobDelimiter`, i.e: `/`, to the `blobPrefix` if it's not there. But we might consider removing that in the future. It'll be a breaking change though, so we might only do it for v3?